aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -17,6 +17,7 @@ pub struct TransactionMetadata {
     pub secondary_signers: Vec<AccountAddress>,
     pub secondary_authentication_keys: Vec<Vec<u8>>,
     pub sequence_number: u64,
+    pub with_gas_payer: bool,
```

Comment:
I don't understand why this isn't an explicit field.

Reply:
I think it's good to work towards an orthogonal primitives, instead of adding special cases for each new hot feature we think of. We have a authenticator for multiple signatures, either we reuse it or we implement a 3rd special cased with potential bugs.